application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,152 @@
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user who triggered the action
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     *
+     * @param attachment an attachment that'll be added to the message
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild,
+            @NotNull Attachment attachment) {
+        writeModAuditLog(title, description, author, timestamp, guild, List.of(attachment));
+    }
+
+    /**
+     * logs an entry in the mod audit log channel.
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user who triggered the action
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild) {
+        writeModAuditLog(title, description, author, timestamp, guild, List.of());
+    }
+
+    /**
+     * used to get the mod audit log channel.
+     *
+     * @param guild the current guild
+     */
+    public static Optional<TextChannel> getModAuditLogChannel(@NotNull Guild guild) {
+        Optional<TextChannel> channel =
+                guild.getTextChannelCache().stream().filter(isAuditLogChannel).findAny();
+        if (channel.isEmpty()) {
+            logger.warn(
+                    "Unable to log moderation events, did not find a mod audit log channel matching the configured pattern '{}' for guild '{}'",
+                    Config.getInstance().getModAuditLogChannelPattern(), guild.getName());
+        }
+        return channel;
+    }
+
+    /**
+     * used to add a file to a message without having an actual file.
+     * 
+     * @param name the name of the file, example: {@code "foo.md"}
+     * @param content the content of the file
+     */
+    public static final record Attachment(@NotNull String name, @NotNull String content) {
+        /**
+         * @return the name of the file. used by JDA methods
+         */
+        public @NotNull String getName() {
+            return name;
+        }
+
+        /**
+         * @return the content of the file as a {@code byte[]}. used by JDA methods
+         */
+        public byte @NotNull [] getContent() {
+            return content.getBytes(StandardCharsets.UTF_8);
+        }
```

Comment:
* The javadoc has the same problems as the other method.
* The name might be misleading and is definitely confusing since the parameter had a different type. Call it `getContentRaw()` instead.
* Javadoc should then be `"Gets the content raw, interpreted as UTF-8."` with `"@return the raw content"`

Reply:
done.